### DISCUSSION TITLE
Dify Sandbox v0.2.6: Install any Python package you want

### DISCUSSION THREAD

**Discussion by MAlj11** [langgenius/dify#8200] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Provide a description of requested docs changes

In the version introduction of v0.7.1, it is mentioned that "Dify Sandbox v0.2.6: Install any Python package you want". Does this mean that I can install the Python package I want to use in the sandbox and then use it directly in the code execution node? Now that I have successfully deployed Dify Sandbox v0.2.6, may I know how to use this feature and if there are any related tutorials

**Comment by crazywoola** [comment]
See https://github.com/langgenius/dify-sandbox/blob/main/FAQ.md

```
var ALLOW_SYSCALLS = []int{
	// file io
	syscall.SYS_NEWFSTATAT, syscall.SYS_IOCTL, syscall.SYS_LSEEK, syscall.SYS_GETDENTS64,
	syscall.SYS_WRITE, syscall.SYS_CLOSE, syscall.SYS_OPENAT, syscall.SYS_READ,
    
	...

	// run numpy required
	syscall.SYS_FSTAT, syscall.SYS_PREAD64, syscall.SYS_MADVISE, syscall.SYS_UNAME,
	syscall.SYS_SCHED_GETAFFINITY, syscall.SYS_MBIND, syscall.SYS_EPOLL_PWAIT, 435,
}
```

https://github.com/langgenius/dify/blob/bbef964eb5caaca6df74b21a596900f31faff677/docker/volumes/sandbox/dependencies/python-requirements.txt Add your packages here and restart the sandbox.

**Reply by MAlj11** [reply]
I added 'mistune' to the python-requirements. txt file, but it doesn't work and prompts that I can't find the package. However, I added 'import mistune' to the prescription. py file and it works fine. May I ask what the reason is? Is it because I used or deployed it incorrectly

**Reply by MAlj11** [reply]
![1](https://github.com/user-attachments/assets/756783fd-91aa-4d3c-8e74-56d7a9bfa94b)

**Reply by amroeser** [reply]
> See https://github.com/langgenius/dify-sandbox/blob/main/FAQ.md
> 
> ```
> var ALLOW_SYSCALLS = []int{
> 	// file io
> 	syscall.SYS_NEWFSTATAT, syscall.SYS_IOCTL, syscall.SYS_LSEEK, syscall.SYS_GETDENTS64,
> 	syscall.SYS_WRITE, syscall.SYS_CLOSE, syscall.SYS_OPENAT, syscall.SYS_READ,
>     
> 	...
> 
> 	// run numpy required
> 	syscall.SYS_FSTAT, syscall.SYS_PREAD64, syscall.SYS_MADVISE, syscall.SYS_UNAME,
> 	syscall.SYS_SCHED_GETAFFINITY, syscall.SYS_MBIND, syscall.SYS_EPOLL_PWAIT, 435,
> }
> ```
> 
> https://github.com/langgenius/dify/blob/bbef964eb5caaca6df74b21a596900f31faff677/docker/volumes/sandbox/dependencies/python-requirements.txt Add your packages here and restart the sandbox.

Hi @crazywoola , Pascal told me to talk to you about my problem. I have installed the package based on the mentioned file, but I can't run simple python code with pandas because it is not whitelisted. The posts in the community did not help me in this case because the sandbox is hosted in a docker container. Is there a way to whitelist specific packages like pandas?

**Reply by DemonXD** [reply]
i already have the same problem, in the contain, i found the /usr/loca/lib/python/..../site-packages had installed the libs i put in the python-requirements.txt, but the actual python interpreter in the sandbox-python/.../site-packages did not install the libs, is there anything i did not check?

**Comment by brightwang** [comment]
I made a video tutorial, you can check it out.
https://www.youtube.com/watch?v=gUGsjRuLAdQ

**Reply by SpiderJack0516** [reply]
I've done as in the video , but opencv and numpy still don't work.
![image](https://github.com/user-attachments/assets/d115d47d-f298-46b0-bad1-547741d53880)


> I made a video tutorial, you can check it out. https://www.youtube.com/watch?v=gUGsjRuLAdQ

**Reply by SpiderJack0516** [reply]
solved by "sudo docker restart docker-sandbox-1"
